sql/core/src/main/scala/org/apache/spark/sql/catalyst/util/V2ExpressionBuilder.scala

```diff
@@ -209,7 +209,12 @@ class V2ExpressionBuilder(e: Expression, isPredicate: Boolean = false) {
       } else {
         None
       }
-    case iff: If => generateExpressionWithName("CASE_WHEN", iff.children)
+    case iff: If =>
```

Comment:
can you check other expressions matched here and see if they may return boolean type?

Reply:
> can you check other expressions matched here and see if they may return boolean type?

there is also `Coalesce`.